applications/zigbee_weather_station/CMakeLists.txt

```diff
@@ -0,0 +1,34 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+cmake_minimum_required(VERSION 3.20.0)
+
+
+if (NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/configuration/${BOARD}")
```

Comment:
This is not a good pattern, cause it will break for users trying to build for a specific board revision.

For example `nrf9160dk_nrf9160_ns@0.14.0`.
Now I know this board is not supported in this sample, but from this PR https://github.com/nrfconnect/sdk-nrf/pull/1637 I see the original intention is to easier allow users to provide add configurations for other boards, and such boards can be using the revision feature.

Reply:
Let's look into this after next upmerge.

Have an idea how this feature can be made generic.